src/udev/udev-rules.c

```diff
@@ -1409,6 +1409,30 @@ static bool tokens_eq(const UdevRuleToken *a, const UdevRuleToken *b) {
                token_type_and_data_eq(a, b);
 }
 
+static bool nulstr_tokens_conflict(const UdevRuleToken *a, const UdevRuleToken *b) {
+        assert(a);
+        assert(b);
+
+        if (!(a->type == b->type &&
+              type_has_nulstr_value(a->type) &&
+              a->op == b->op &&
+              a->op == OP_MATCH &&
+              a->match_type == b->match_type &&
+              a->match_type == MATCH_TYPE_PLAIN &&
+              a->attr_subst_type == b->attr_subst_type &&
+              a->attr_match_remove_trailing_whitespace == b->attr_match_remove_trailing_whitespace &&
+              token_type_and_data_eq(a, b)))
+                return false;
+
+        const char *a_value = a->value;
+        const char *b_value = b->value;
+        NULSTR_FOREACH(i, a_value)
+                if (nulstr_contains(b_value, i))
```

Comment:
Why not directly use `a->value` or `b->value` in `NULSTR_FOREACH()` or `nulstr_contains()`?

Reply:
I must have confused the type of `value` with the type of `data`.